Fix for environment setup with PyPI package build action

d33bs opened the pull request:
<!-- _modified from [EmbeddedArtistry](https://embeddedartistry.com/blog/2017/08/04/a-github-pull-request-template-for-your-projects/)_
_referenced with modifications from [pycytominer](https://github.com/cytomining/pycytominer/blob/master/.github/PULL_REQUEST_TEMPLATE.md)_ -->

# Description

This fixes a bug with the environment setup for PyPI package builds through GitHub actions exhibited [here](https://github.com/WayScience/coSMicQC/actions/runs/9368841020/job/25791866149). This was a fault on my part in attempting to reuse code across projects (apologies!).

Thanks for any thoughts from review here!

## What kind of change(s) are included?

- [ ] Feature (adds or updates new capabilities)
- [x] Bug fix (fixes an issue).
- [ ] Enhancement (adds functionality).
- [ ] Breaking change (these changes would cause existing functionality to not work as expected).

# Checklist

Please ensure that all boxes are checked before indicating that this pull request is ready for review.

- [x] I have read and followed the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] I have searched for existing content to ensure this is not a duplicate.
- [x] I have performed a self-review of these additions (including spelling, grammar, and related).
- [x] These changes pass all pre-commit checks.
- [ ] I have added comments to my code to help provide understanding
- [ ] I have added a test which covers the code changes found within this PR
- [x] I have deleted all non-relevant text in this pull request template.

jenna-tomkinson reviewed (approved):
LGTM! For my own understanding, what was the issue/bug?

d33bs commented:
Thanks @jenna-tomkinson ! 

> For my own understanding, what was the issue/bug?

We rely on [`poetry build`](https://python-poetry.org/docs/cli/#build) to help collect the Python package materials for distribution on PyPI (which requires special organization of the files for installation from that platform). The GitHub Actions workflow associated with this is a container which doesn't know about Python, Poetry, or our package state by default. Thus, in order to use `poetry build` we need to install Python, Poetry, and our package dependencies (from `poetry install`) in order for the container to know how to build the package for PyPI. The earlier code which was intended to perform this is a reused block from another project which I recycled but wasn't thorough enough about checking before submitting a PR (bad form on my part, sorry!). We don't yet have the related capabilities from that block of code (and I'm not certain they're necessary yet here). So to fix the bug, I removed this unnecessary / non-existent capability, and added the related installation steps which are needed. These align with other Python and Poetry installation steps which are found for example in https://github.com/WayScience/coSMicQC/blob/main/.github/workflows/run-tests.yml .
